runtime/observability/telemetry-api/common/src/aws/smithy/kotlin/runtime/telemetry/TelemetryProvider.kt

```diff
@@ -25,24 +26,29 @@ public interface TelemetryProvider {
     /**
      * Get the [TracerProvider] used to create new [aws.smithy.kotlin.runtime.telemetry.trace.Tracer] instances
      */
+    @ExperimentalApi
     public val tracerProvider: TracerProvider
 
     /**
      * Get the [MeterProvider] used to create new [aws.smithy.kotlin.runtime.telemetry.metrics.Meter] instances
      */
+    @ExperimentalApi
     public val meterProvider: MeterProvider
 
     /**
      * Get the [LoggerProvider] used to create new [aws.smithy.kotlin.runtime.telemetry.logging.Logger] instances
      */
+    @ExperimentalApi
     public val loggerProvider: LoggerProvider
 
     /**
      * Get the [ContextManager] used to get the current [Context]
      */
+    @ExperimentalApi
     public val contextManager: ContextManager
 }
 
+@OptIn(ExperimentalApi::class)
```

Comment:
**Question**: Shouldn't we opt into this annotation globally [the way we do with `@InternalApi`](https://github.com/awslabs/smithy-kotlin/blob/3e1d8618b0090b218461cffdcce20619d6b4119b/runtime/auth/aws-signing-default/build.gradle.kts#L27C1-L29)?

Reply:
I decided not to because I wanted to see how far reaching the API becomes. Turns out there weren't that many places that need it.